### ISSUE TITLE
Generate own model 

### ISSUE THREAD

**Issue by deblockt** [springfox/springfox#2043]
Please take the time to search the repository, if your question has already been asked or answered.

- [x] What version of the library are you using? Is it the latest version? The latest released version is [ ![Download](https://api.bintray.com/packages/springfox/maven-repo/springfox/images/download.svg) ](https://bintray.com/springfox/maven-repo/springfox/_latestVersion)

What kind of issue is this?

 - [x] Question. Is this a question about how to do a certain thing? 

I need to create my own model from my Entity. My entities are simple DAO, and i need to generate jsonApi json from my Entity.

I have successfully done this feature, but to do that I must override `DefaultModelProvider` class.
I think this is not a good idea to do that.

I have try to generate my model with `ModelBuilderPlugin` but with this plugin, I can't add model dependencies. And my json api model have serveral model depedencies. 

I have try to use custom `ModelDependencyProvider` but method `dependentModels` return Set of `ResolvedType` but my model is "fictif" and it's not associated to a `Java type` 

There are possibility to add model dependency on `ModelBuilderPlugin`or with other plugin ?

Exemple : 
model:
``` Scala
@JsonApi(`type`="demo")
class Demo() {

  @BeanProperty
  @JsonApiId
  var apiId: UUID = _

  @BeanProperty
  var label: String = _

  @BeanProperty
  @JsonApiRelationShip
  var writer: Person = _
}
```

must generate model for this json structure :
```json
{
  "data": {
    "attributes": {
      "label": "string"
    },
    "id": "string",
    "relationships": {
      "writer": {
        "data": {
          "id": "string",
          "type": "person"
        }
      }
    },
    "type": "demo"
  }
}
```

**Comment by dilipkrish** [comment]
Yeah it relies on real types. Do you have any ideas?

**Comment by dilipkrish** [comment]
This is resolved via #2311, where we create a synthetic type to represent `Resources<T>` e.g. [`ResourcesModelProvider`](https://github.com/springfox/springfox/blob/master/springfox-data-rest/src/main/java/springfox/documentation/spring/data/rest/schema/ResourcesModelProvider.java).
